sphinx/source/conf.py

```diff
@@ -275,3 +270,76 @@
     'pandas': ('http://pandas.pydata.org/pandas-docs/stable/', None),
     'numpy': ('https://docs.scipy.org/doc/numpy/', None)
 }
+
+
+import os
+import subprocess
+
+# Do not use Bokeh CDNs to serve JS files when building latest version
+# of docs (and/or unreleased versions). We want to serve the compiled
+# JS files from Read the Docs servers itself to match the
+# documentation with the JS version. Although, when using a tagged
+# release, we want to use Bokeh CDN.
+def _get_current_commit():
+    result = subprocess.run(
+        ['git', 'rev-parse', '--verify', 'HEAD'],
+        stdout=subprocess.PIPE,
+    )
+    return result.stdout.decode('utf8').strip()
+
+def is_tagged_version():
+    commit = _get_current_commit()
+    print('Bokeh commit: {commit}'.format(commit=commit))
+    result = subprocess.run(
+        ['git', 'tag', '--points-at', commit],
+        stdout=subprocess.PIPE,
+    )
+    tagged = result.stdout.decode('utf8').strip()
+    print('Bokeh tagged version: {tagged}'.format(tagged=tagged))
+    return bool(tagged)
+
+if os.environ.get('READTHEDOCS') == 'True':
+    readthedocs_version = os.environ.get('READTHEDOCS_VERSION')
+    if is_tagged_version():
+        # Use Bokeh Resources from CDN
+        os.environ['BOKEH_DOCS_CDN'] = readthedocs_version
+    else:
+        # Use Bokeh Resources from Read the Docs
+        os.environ['BOKEH_DOCS_CDN'] = 'test:{readthedocs_version}'.format(
+            readthedocs_version=readthedocs_version,
+        )
+
+    bokeh_docs_cdn = os.environ.get('BOKEH_DOCS_CDN')
+    print('BOKEH_DOCS_CDN: {bokeh_docs_cdn}'.format(
+        bokeh_docs_cdn=bokeh_docs_cdn,
+    ))
+
+    # Set the proper ``BOKEH_DOCS_VERSION`` based on the version being
+    # built on Read the Docs
+    os.environ['BOKEH_DOCS_VERSION'] = readthedocs_version
+    print('BOKEH_DOCS_VERSION: {readthedocs_version}'.format(
+        readthedocs_version=readthedocs_version,
+    ))
+
+    build_js = os.environ.get('BOKEH_BUILD_JS')
+    print('BOKEH_BUILD_JS: {build_js}'.format(build_js=build_js))
+
+    # Download all the sample data to plot
+    import bokeh
+    bokeh.sampledata.download(progress=False)
+
+    # Create the proper structure to serve CSS and JS files compiled by
+    # Bokeh. We can't use ``html_static_path`` here because the URLs are
+    # hardcoded to ``static/js`` instead of ``_static/js``, etc
+    os.system('mkdir -p bokehJSbuildstatic')
+    os.system('cd bokehJSbuildstatic; ln -s ../../../bokehjs/build static')
```

Comment:
I think this would break on Windows, e.g so perhaps anything like this should be conditioned on the "is this RTD?" environment var